x-pack/legacy/plugins/infra/public/containers/logs/log_filter/use_log_filter_url_state.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useContext, useEffect } from 'react';
+import * as rt from 'io-ts';
+import { LogFilterState } from './log_filter_state';
+import { replaceStateKeyInQueryString } from '../../../utils/url_state';
+import { useUrlState } from '../../../utils/use_url_state';
+import { getEncodeDecodeFromRT } from '../../../utils/validate_url_rt';
+
+const logFilterRT = rt.union([
+  rt.type({
+    kind: rt.literal('kuery'),
+    expression: rt.string,
+  }),
+  rt.undefined,
+]);
+type LogFilterUrlState = rt.TypeOf<typeof logFilterRT>;
+
+const LOG_FILTER_URL_STATE_KEY = 'logFilter';
+
+export const useLogFilterUrlState = () => {
+  const { filterQueryAsKuery, applyLogFilterQuery } = useContext(LogFilterState.Context);
+  const [logFilterUrlState, setLogFilterUrlState] = useUrlState({
+    defaultState: filterQueryAsKuery,
+    urlStateKey: LOG_FILTER_URL_STATE_KEY,
+    writeDefaultState: true,
+    ...getEncodeDecodeFromRT(logFilterRT),
+  });
+
+  /* eslint-disable react-hooks/exhaustive-deps */
+  useEffect(() => {
+    if (logFilterUrlState && filterQueryAsKuery?.expression !== logFilterUrlState.expression) {
+      applyLogFilterQuery(logFilterUrlState.expression);
+    }
+  }, [logFilterUrlState]);
```

Comment:
No, we only want to run this effect when the URL changes. `applyLogFilterQuery` changes the value of `filterQueryAsKuery`, so best case scenario it would run a second time for no reason, worst case it would lead to an infinite loop if React decides to update certain state values in an unpredictable order.

Reply:
Actually, sorry, this code was from https://github.com/elastic/kibana/pull/52656 but then reverted, and seems to have reappeared due to a rebasing issue. Disregard for now. We're going to look into `useUrlState` in a separate issue.